### DISCUSSION TITLE
Binding to the navigation stack

### DISCUSSION THREAD

**Discussion by D-Diyare** [MvvmCross/MvvmCross#4119] [Q&A]
Hello, Thanks for the great framework.

I was wondering if it is possible to bind to the navigation stack?, for example: let's say we have 3 view models,
- ViewModel A
- ViewModel B
- ViewModel C

The Application starts at ViewModel A, then navigate to ViewModel B then ViewModel C, its all store in the navigation stack as we can go back by closing the current view model, now how to know the stack items of navigation? I would like to show the navigation stack as a breadcrumb.

Best Regards.

**Comment by Hackmodford** [comment]
Have you considered subclassing the NavigationService to do this?

You could create your own implementations of `Navigate` and keep track of the view models in your subclass.

**Reply by D-Diyare** [reply]
Nice trick, I try to give it a try soon
